app/models/spree/order.rb

```diff
@@ -434,6 +434,7 @@ def empty!
       payments.clear
       shipments.destroy_all
       restart_checkout_flow if state == "payment"
+      restart_checkout_flow if state == "confirmation"
```

Comment:
Since we are talking about this already, I would probably do this:
```suggestion
      restart_checkout_flow if state.in?(["payment", "confirmation"])
```

Reply:
In terms of readability, id pick this over mine